edb/server/mng_port/edgecon.pyx

```diff
@@ -824,6 +867,10 @@ cdef class EdgeConnection:
             self.debug_print('SIMPLE QUERY', eql)
 
         if self.dbview.in_tx_error():
+            if not (allow_capabilities & enums.Capability.TRANSACTION):
```

Comment:
Which scenario triggers this?  Is this check really needed?

Reply:
Perhaps something like this:
```
with conn.try_transaction() as tx:
    try:
        do_something()
    except:
        tx.query("ROLLBACK")  # << here we disable transaction statements, per RFC1004
```